src/compiler/checker.ts

```diff
@@ -19043,26 +19043,48 @@ namespace ts {
                 diagnostics.add(getTypeArgumentArityError(node, signatures, typeArguments));
             }
             else if (args) {
-                let min = Number.POSITIVE_INFINITY;
-                let max = Number.NEGATIVE_INFINITY;

```

Comment:
I think you should revert all the first section of this change and add two more variables. This worked for me in simple tests:

```ts
let belowArgCount = Number.NEGATIVE_INFINITY
let aboveArgCount = Number.POSITIVE_INFINITY
for (const sig of signatures) {
    minCount = getMinArgumentCount(sig)
    maxCount = getParameterCount(sig)
    if (minCount < argCount && argCount - minCount < argCount - belowArgCount) belowArgCount = minCount
    if (argCount < maxCount && maxCount - argCount < aboveArgCount - argCount) aboveArgCount = maxCount
    min = Math.min(min, minCount)
    max = Math.max(max, maxCount)
}
```

(Note that you'll have to calculate argCount earlier.) Then you can use belowArgCount/aboveArgCount in the new error reporting branch you added in the second half of this change.